### DISCUSSION TITLE
purpose of "config set"

### DISCUSSION THREAD

**Discussion by mvolkmann** [nushell/nushell#3156] [Q&A] [answered]
What is the purpose of the "config set" command?
The docs show `config set path $nu.path` and `config set env $nu.env`, but it's unclear to me what those do.
It seems that `echo $nu.path` gives the same output regardless of whether I run `config set path $nu.path`.

**Comment by mvolkmann** [comment, marked as answer]
I see now that `config set path $nu.path` writes the value of `$nu.path` into my `config.toml` file.
But why would I want that?
Similarly, I see that `config set env $nu.env` writes all the current environment variables into my `config.toml` file.
Why would I want that?

**Reply by fdncred** [reply]
> Why would I want that?

For the same reason bash/zsh/sh/windows allow you to set environment variables and entries in your path.

**Reply by mvolkmann** [reply]
It seems like if I don't do that and those values are not in my Nushell `config.toml` file, the `which` command still uses the PATH in the parent shell to find things and all the environment variables in the parent shell are still available in `$nu.env`. So it's not clear to me what additional benefit I get from having `path` and all the environment variables set in `config.toml`. Is the point to not rely on the parent shell for those?

**Reply by fdncred** [reply]
1. you can set env and paths specific to nushell
2. if nushell is your login shell there is no parent shell to inherit from

**Reply by mvolkmann** [reply]
Thanks Darren!

**Reply by mvolkmann** [reply]
This question is still marked as "unanswered", but I'm satisfied with the answer and want to mark it as answered. How can I do that?

**Reply by fdncred** [reply]
i pushed the checkmark button on the top right of this thread
![image](https://user-images.githubusercontent.com/343840/111078184-50db4000-84c2-11eb-8b5e-04c8c0ca4148.png)
